src/geometry/3d_intersection_of_planes.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+typedef pair<Point3d,bool> P3db;
+
+/* 
+[*] Input:
+    ２つの平面 pl1, pl2
+[*] Output:
+    ２つの平面が交線をもつ場合 -> first:交線上の任意の１点, second:true
+　　 　　　 交線を持たない場合 -> first:empty           , second:false
+ */
+P3db intersectPlPl(const Plane3d &pl1,const Plane3d &pl2){
+  Vector3d v = cross(pl1.normal_vector,pl2.normal_vector);
+  if( !equals(v.x,0.0) ){
+    return P3db(Point3d(0,
+                        (pl1.d*pl2.normal_vector.z-pl2.d*pl1.normal_vector.z)/v.x,
+                        (pl1.d*pl2.normal_vector.y-pl2.d*pl1.normal_vector.y)/(-v.x)),true);
+  }
+  if( !equals(v.y,0.0) ){
+    return P3db(Point3d((pl1.d*pl2.normal_vector.z-pl2.d*pl1.normal_vector.z)/(-v.y),
+                         0,
+                        (pl1.d*pl2.normal_vector.x-pl2.d*pl1.normal_vector.x)/v.y),true);
+  }
+  if( !equals(v.z,0.0) ){
+    return P3db(Point3d((pl1.d*pl2.normal_vector.y-pl2.d*pl1.normal_vector.y)/v.z,
+                        (pl1.d*pl2.normal_vector.x-pl2.d*pl1.normal_vector.x)/(-v.z),
+                        0),true);
+  }
```

Comment:
```suggestion
  if( !equals(v.x,0.0) ){
    Point3d p(0,
              (pl1.d*pl2.normal_vector.z-pl2.d*pl1.normal_vector.z)/v.x,
              (pl1.d*pl2.normal_vector.y-pl2.d*pl1.normal_vector.y)/(-v.x));
    return P3db(p, true);
  }
  if( !equals(v.y,0.0) ){
    Point3d p((pl1.d*pl2.normal_vector.z-pl2.d*pl1.normal_vector.z)/(-v.y),
              0,
              (pl1.d*pl2.normal_vector.x-pl2.d*pl1.normal_vector.x)/v.y);
    return P3db(p,true);
  }
  if( !equals(v.z,0.0) ){
    Point3d p((pl1.d*pl2.normal_vector.y-pl2.d*pl1.normal_vector.y)/v.z,
              (pl1.d*pl2.normal_vector.x-pl2.d*pl1.normal_vector.x)/(-v.z),
              0);
    return P3db(p,true);
  }
```

これは微妙なところなんですが、交線上の点pをreturnから分離したほうが横幅を抑えられるのと解釈がしやすいかなと思いました
完全に好みなので単純に提案です 🙇